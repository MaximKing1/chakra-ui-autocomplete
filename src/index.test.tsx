import { CUIMultipleAutocomplete } from './index'

import React from 'react'
import renderer from 'react-test-renderer'
import { Text, Flex, Heading, Avatar } from '@chakra-ui/react'

export interface Item {
  label: string
  value: string
}
const countries = [
  { value: 'ghana', label: 'Ghana' },
  { value: 'nigeria', label: 'Nigeria' },
  { value: 'kenya', label: 'Kenya' },
  { value: 'southAfrica', label: 'South Africa' },
  { value: 'unitedStates', label: 'United States' },
  { value: 'canada', label: 'Canada' },
  { value: 'germany', label: 'Germany' }
]

export default function MultiSelect() {
  const [pickerItems, setPickerItems] = React.useState(countries)
  const [selectedItems, setSelectedItems] = React.useState<Item[]>([])

  const handleCreateItem = (item: Item) => {
    setPickerItems((curr) => [...curr, item])
    setSelectedItems((curr) => [...curr, item])
  }

  const handleSelectedItemsChange = (selectedItems?: Item[]) => {
    if (selectedItems) {
      setSelectedItems(selectedItems)
    }
  }

  const customRender = <T extends Item>(selected: T) => {
    return (
      <Flex flexDir='row' alignItems='center'>
        <Avatar mr={2} size='sm' name={selected.label} />
        <Text>{selected.label}</Text>
      </Flex>
    )
  }

  return (
    <Flex
      flexWrap='wrap'
      px={6}
      pt={12}
      height='100vh'
      bg='rgba(247,250,252)'
      justify='center'
      borderRight='1px solid #ddd'
    >
      <Flex minW='25rem' w='25rem' flexDirection='column'>
        <Heading mb={3} as='h1' size='md'>
          Basic Multiple Example with Style Props
        </Heading>

        <CUIMultipleAutocomplete
          tagStyleProps={{
            rounded: 'full'
          }}
          labelStyleProps={{
            color: 'rebeccapurple'
          }}
          label='Choose preferred work locations'
          placeholder='Type a Country'
          onCreateItem={handleCreateItem}
          items={pickerItems}
          selectedItems={selectedItems}
          onSelectedItemsChange={(changes) =>
            handleSelectedItemsChange(changes.selectedItems)
          }
        />

        <Flex mt={6} />
        <Heading mb={3} as='h1' size='md'>
          Basic Multiple Example with Custom Renderer
        </Heading>
        <CUIMultipleAutocomplete
          tagStyleProps={{
            rounded: 'full'
          }}
          label='Choose preferred work locations'
          placeholder='Type a Country'
          onCreateItem={handleCreateItem}
          items={pickerItems}
          itemRenderer={customRender}
          selectedItems={selectedItems}
          onSelectedItemsChange={(changes) =>
            handleSelectedItemsChange(changes.selectedItems)
          }
        />
      </Flex>
    </Flex>
  )
}

describe('ChakraUI Autocomplete ', () => {
  test('it renders', () => {
    const tree = renderer.create(<MultiSelect />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
