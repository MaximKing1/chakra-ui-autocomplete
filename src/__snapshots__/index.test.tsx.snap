// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChakraUI Autocomplete  it renders 1`] = `
<div
  className="css-1010j8n"
>
  <div
    className="css-vy8xsp"
  >
    <h1
      className="chakra-heading css-h82jag"
    >
      Basic Example with Style Props
    </h1>
    <div
      className="chakra-stack css-1pifhv5"
    >
      <label
        className="chakra-form__label css-1l2u59i"
        htmlFor="downshift-0-input"
        id="downshift-0-label"
      >
        Choose preferred work locations
      </label>
      <div
        className="chakra-stack css-c7vfsi"
      />
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-owns="downshift-0-menu"
        className="chakra-stack css-loe1j3"
        role="combobox"
      >
        <input
          aria-autocomplete="list"
          aria-controls="downshift-0-menu"
          aria-labelledby="downshift-0-label"
          autoComplete="off"
          className="chakra-input css-0"
          id="downshift-0-input"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Type a Country"
          value=""
        />
        <button
          aria-label="toggle menu"
          className="chakra-button css-oqfhsh"
          id="downshift-0-toggle-button"
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            className="chakra-icon css-onkibi"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        className="css-vm1uz1"
      >
        <ul
          aria-labelledby="downshift-0-label"
          className="css-1jrokhb"
          id="downshift-0-menu"
          onMouseLeave={[Function]}
          role="listbox"
        />
      </div>
    </div>
    <div
      className="css-kkyj92"
    />
    <h1
      className="chakra-heading css-h82jag"
    >
      Basic Example with Custom Renderer
    </h1>
    <div
      className="chakra-stack css-1pifhv5"
    >
      <label
        className="chakra-form__label css-uvho5g"
        htmlFor="downshift-1-input"
        id="downshift-1-label"
      >
        Choose preferred work locations
      </label>
      <div
        className="chakra-stack css-c7vfsi"
      />
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-owns="downshift-1-menu"
        className="chakra-stack css-loe1j3"
        role="combobox"
      >
        <input
          aria-autocomplete="list"
          aria-controls="downshift-1-menu"
          aria-labelledby="downshift-1-label"
          autoComplete="off"
          className="chakra-input css-0"
          id="downshift-1-input"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Type a Country"
          value=""
        />
        <button
          aria-label="toggle menu"
          className="chakra-button css-oqfhsh"
          id="downshift-1-toggle-button"
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        >
          <svg
            className="chakra-icon css-onkibi"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        className="css-vm1uz1"
      >
        <ul
          aria-labelledby="downshift-1-label"
          className="css-1jrokhb"
          id="downshift-1-menu"
          onMouseLeave={[Function]}
          role="listbox"
        />
      </div>
    </div>
  </div>
</div>
`;
